File: 2133-check-if-every-row-and-column-contains-all-numbers/2133-check-if-every-row-and-column-contains-all-numbers.java
class Solution {
    public boolean checkValid(int[][] matrix) {
        
        for(int i=0; i<matrix.length;i++)
        {
            HashSet<Integer> hs= new HashSet<>();
            
            for(int j=0; j<matrix.length; j++)
            {
                hs.add(matrix[i][j]);
            }
            if(hs.size()!=matrix.length)
            {
                return false;
            }
        }
        
        for(int i=0; i<matrix.length; i++)
        {
            HashSet<Integer> hs = new HashSet<>();
            for(int j=0; j<matrix.length; j++)
            {
                hs.add(matrix[j][i]);
            }
            if(hs.size()!=matrix.length)
            {
                return false;
            }
        }
        return true;
    }
}

File: 2133-check-if-every-row-and-column-contains-all-numbers/2133-check-if-every-row-and-column-contains-all-numbers.cpp
class Solution {
public:
    bool checkValid(vector<vector<int>>& matrix) {
        
        int n=matrix.size();
        
        unordered_set<int> row;
        unordered_set<int> col;
        
        for(int i=0; i<matrix.size();i++)
        {
            for(int j=0; j<matrix.size(); j++)
            {
                row.insert(matrix[i][j]);
                col.insert(matrix[j][i]);
            }
            if(row.size()!=matrix.size() || col.size()!=matrix.size())
            {
                return false;
            }
            row.clear();
            col.clear();
        }
        
        return true;
    }
};